### ISSUE TITLE
Bug: Upgrading to v9 seems extremely painful - getting tons of linting errors not reported before upgrade

### ISSUE THREAD

**Issue by frank-topel-dbi** [eslint/eslint#19316]
### Environment

Node version: v22.12.0
npm version: v11.0.0
Local ESLint version: v9.17.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 24.2.0

### What parser are you using?

babelParser

### What did you do?

I tried upgrading to ESLint 9 (from the latest 8.x), and migrating the `.eslintrc.json` via the migration tool `npx @eslint/migrate-config .eslintrc.json`, and also updating the plugins `"eslint-plugin-json": "^4.0.1",  "eslint-plugin-unused-imports": "^4.1.4",`, **and** moving the `ignore` section from `package.json` to the newly created `eslint.config.mjs`.

<details>
<summary>Configuration</summary>

```
import unusedImports from 'eslint-plugin-unused-imports';
import json from 'eslint-plugin-json';
import globals from 'globals';
import babelParser from '@babel/eslint-parser';
import path from 'node:path';
import { fileURLToPath } from 'node:url';
import js from '@eslint/js';
import { FlatCompat } from '@eslint/eslintrc';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
  baseDirectory: __dirname,
  recommendedConfig: js.configs.recommended,
  allConfig: js.configs.all,
});

export default [
  ...compat.extends('eslint:recommended', 'plugin:json/recommended-with-comments-legacy'),
  {
    plugins: {
      'unused-imports': unusedImports,
      json,
    },

    languageOptions: {
      globals: {
        ...globals.browser,
        ...globals.node,
        ...globals.jest,
        __dirname: true,
      },

      parser: babelParser,
      ecmaVersion: 'latest',
      sourceType: 'module',

      parserOptions: {
        requireConfigFile: false,

        babelOptions: {
          plugins: ['@babel/plugin-syntax-import-assertions'],
        },
      },
    },

    rules: {
      'no-console': [
        'error',
        {
          allow: ['warn', 'error'],
        },
      ],

      'no-unused-vars': 'off',
      'unused-imports/no-unused-imports': 'off',

      'unused-imports/no-unused-vars': [
        'error',
        {
          vars: 'all',
          varsIgnorePattern: '^_',
          args: 'after-used',
          argsIgnorePattern: '^_',
        },
      ],

      'json/*': 'error',

      'prefer-const': [
        'error',
        {
          destructuring: 'all',
        },
      ],

      'no-param-reassign': 'warn',
    },
    ignores: [
      '**/dist/**/*',
      '**/node_modules/**/*',
      '**/build/**/*',
      '**/dev/**/*',
      '.*',
      '.*rc.json',
      'package.json',
      'package-lock.json',
      'node_modules',
      'node_scripts',
      '*.d.ts',
      '*.sample.js',
      '**/vite.config.js',
      '**/playwright-report/**/*',
    ],
  },
];

```
</details>

### What did you expect to happen?

I was expecting ESLint to not report any problems due to the rules, other than those added or strictified in v9. I have read the breaking changes log for v9, and invested some time into reading the migration guide.

### What actually happened?

I'm getting a gazillion ESLint errors which weren't present before upgrading:

[eslint.log.txt](https://github.com/user-attachments/files/18321698/eslint.log.txt)

Many of them related to unused private members (which aren't unused in code), unknown global variables (despite listing `globals.browser`) etc.

### Link to Minimal Reproducible Example

https://example.com

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I know upgrading to a new major version of essential tooling is sometimes painful. Due to the sheer amount of errors I'm honestly lost here. I'd be very grateful for advice, otherwise I fear we will be stuck with v8 forever.

Since the code base lives in a private and quite big company repo, I cannot provide a reproducer. I am hoping for someone to point me in the right direction to try and solve this.

**Comment by mdjermanovic** [comment]
> ```
> ignores: [
>       '**/dist/**/*',
>       '**/node_modules/**/*',
>       '**/build/**/*',
>       '**/dev/**/*',
>       '.*',
>       '.*rc.json',
>       'package.json',
>       'package-lock.json',
>       'node_modules',
>       'node_scripts',
>       '*.d.ts',
>       '*.sample.js',
>       '**/vite.config.js',
>       '**/playwright-report/**/*',
>     ],
> ```

If you want ESLint to ignore these files, you should extract `ignores` into a separate config objects with no other keys.

https://eslint.org/docs/latest/use/configure/ignore

By taking a quick look at your log file, seems that most (if not all) errors are in files that should be ignored, so this will probably fix the problem. If you're still getting errors in files that should not be ignored, please provide concrete code examples.

**Comment by frank-topel-dbi** [comment]
That reduces the number of errors down to a reasonable number of 18, which is a number I feel I can check. Will come back as soon as I'm done.

**Comment by frank-topel-dbi** [comment]
In one of the files, I get reported an "unused private variable" report, despite using it in the constructor.

```js
class BaseButton extends Button {

  /** @type {ElementInternals} */
  #internals;

 /* ... */

  constructor() {
    super();
    this.#internals = this.attachInternals();
  }
}
```

Apparently just assigning a value to make it a form-associated element is not sufficient usage?

**Comment by mdjermanovic** [comment]
Yes, the `no-unused-private-class-members` rule reports private class members that are assigned but never read as unused because they are effectively unused. You can either remove them or use an [eslint-disable](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1) comment to keep them until they become used in the code.

**Comment by frank-topel-dbi** [comment]
I am closing this since the remaining errors appear to be valid reports. Still unsure why these didn't show up before the upgrade, worth fixing nonetheless.

@mdjermanovic Many thanks again for your help! <3
